### DISCUSSION TITLE
x-collapse should use overflow:clip with optional overflow-clip-margin

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#3948] [3. Feature ideas]
Currently, x-collapse is using `overflow:hidden` which cuts off any horizontal shadow. If it would be using `overflow:clip` with an additional `overflow-clip-margin` (which should be configurable), the horizontal shadow would persist.

Merry Xmas 🎄

**Comment by grey4owl** [comment]
I made a PR (https://github.com/alpinejs/alpine/pull/3949) with improvements and added `overflow: clip` instead of `hidden`. You can add `overflow-clip-margin` using CSS to the given element with the `x-collapse` attribute..
